Issues with alert manager.

<!-- Please use this template while reporting an issue and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**Organization Name**:

<!--This information is optional, but you are highly encourage to leave this reference info for us to know better about the context.!-->

**Short summary about the issue/question**:
If the user uses a self-signed certificate, the alert-manager will be fail.
**Brief what process you are following**:

<!--deployment related issues
Please fill this for deployment related issues:
- Operating type: Initial deployment / upgrading / operating etc.
- Brief what deployment process you are following -->

config file:

```
pylon:
    port: 80
    uri: "http://10.10.10.10:80"
    ssl:
      port: 8443
      crt_name: 10.10.10.10.crt
      crt_path: /cluster-configuration/ssl/10.10.10.10.crt
      key_name: 10.10.10.10.key
      key_path: /cluster-configuration/ssl/10.10.10.10.key
```


```
alert-manager:
#   port: 9093 # optional, do not modify this if you do not want to change the port alert-manager is listening on
  pai-bearer-token: '*********************'
  alert-handler: # alert-handler will only be enabled when this field is not empty
#     port: 9095 # optional, do not modify this if you do not want to change the port alert-handler is listening on
#     log-level: "info" # optional
    email-configs: # email-notification will only be enabled when this field is not empty
      admin-receiver: ********@163.com
      smtp-host: smtp.163.com
      smtp-port: 465
      smtp-from: ************@163.com
      smtp-auth-username: *************
      smtp-auth-password: *****************
  cluster-utilization: # cluster-utilization is a k8s CronJob which reports the GPU utilization of the cluster
    # for schedule syntax, refer to https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax
    schedule: "* * * * *" # daily report at UTC 00:00
```


logs:

```
user@hostname:~$ sudo docker logs k8s_cluster-utilization_cluster-utilization-1636635840-s4lxj_default_559d12d2-a499-4327-9427-efadb0447901_3
2021-11-11 13:05:22,502 - INFO - send_alert.py:167 - Collecting cluster usage info...
2021-11-11 13:05:22,507 - DEBUG - connectionpool.py:943 - Starting new HTTPS connection (1): 10.10.10.10:8443
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 978, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 371, in connect
    ssl_context=context,
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/ssl_.py", line 399, in ssl_wrap_socket
    return context.wrap_socket(sock)
  File "/usr/local/lib/python3.7/ssl.py", line 423, in wrap_socket
    session=session
  File "/usr/local/lib/python3.7/ssl.py", line 870, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.7/ssl.py", line 1139, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1076)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 727, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 446, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='10.10.10.10', port=8443): Max retries exceeded with url: /prometheus/api/v1/query?query=avg%28avg_over_time%28nvidiasmi_utilization_gpu%5B7d%5D%29%29 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1076)')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "send_alert.py", line 269, in <module>
    main()
  File "send_alert.py", line 258, in main
    cluster_usage, job_usage, user_usage = collect_metrics(PAI_URI)
  File "send_alert.py", line 30, in wrapper
    return func(*args, **kwargs)
  File "send_alert.py", line 168, in collect_metrics
    resp = requests.get(query_url, params={"query": CLUSTER_QUERY_STRING})
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='10.10.10.10', port=8443): Max retries exceeded with url: /prometheus/api/v1/query?query=avg%28avg_over_time%28nvidiasmi_utilization_gpu%5B7d%5D%29%29 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1076)')))
```



<!--User job related issues
GitHub is not the right place for support requests.
If you're looking for help, check [Stack Overflow](https://stackoverflow.com/questions/tagged/openpai) and the [troubleshooting guide](https://github.com/Microsoft/pai/blob/master/docs/user/troubleshooting_job.md and https://github.com/Microsoft/pai/blob/master/docs/job_tutorial.md#how-to-debug-a-job).
-->

**How to reproduce it**:

<!--Fill the following information if your issue need diagnostic support from the team, as minimally and precisely as possible!-->

**OpenPAI Environment**:
- OpenPAI version: v1.8.0
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): ubuntu 18.04 LTS
- Kernel (e.g. `uname -a`):
- Hardware (e.g. core number, memory size, storage size, GPU type etc.):
- Others:

**Anything else we need to know**:

The error occurred in this line:

https://github.com/microsoft/pai/blob/48394053996bd042a8398fa5d4cfa1f85822a564/src/alert-manager/src/cluster-utilization/send_alert.py#L168

Maybe we should mount the public key to the cluster-utilization container, and then modify this line and other places with similar processes as follows:
```
    resp = requests.get(query_url, params={"query": CLUSTER_QUERY_STRING}, verify='/path/to/public_key.pem')
```

```
user@hostname:~$ sudo docker logs k8s_alerthandler_alertmanager-584c4fcb59-7qft6_default_d47c392f-5a65-4905-b4b7-b0edf08ebb35_0

> pai-alert-handler@0.0.0 start /usr/src/app
> node index.js

2021-11-18T13:33:50.864Z [INFO] alert-handler listening at http://localhost:9095 
2021-11-18T13:35:54.208Z [INFO] alert-handler received `send-email-to-admin` post request from alert-manager. 
2021-11-18T13:36:15.388Z [ERROR] alert-handler failed to send email to admin 
meta = {
  "message": "Unexpected socket close",
  "stack": "Error: Unexpected socket close\n    at Timeout.setTimeout (/usr/src/app/node_modules/nodemailer/lib/smtp-transport/index.js:189:31)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at unrefdHandle (timers.js:520:7)\n    at Timer.processTimers (timers.js:222:12)"
}
2021-11-18T13:36:15.421Z [INFO] alert-handler received `send-email-to-admin` post request from alert-manager. 
2021-11-18T13:36:36.544Z [ERROR] alert-handler failed to send email to admin 
meta = {
  "message": "Unexpected socket close",
  "stack": "Error: Unexpected socket close\n    at Timeout.setTimeout (/usr/src/app/node_modules/nodemailer/lib/smtp-transport/index.js:189:31)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at unrefdHandle (timers.js:520:7)\n    at Timer.processTimers (timers.js:222:12)"
}
2021-11-18T13:36:36.563Z [INFO] alert-handler received `send-email-to-admin` post request from alert-manager. 
2021-11-18T13:36:57.662Z [ERROR] alert-handler failed to send email to admin 
meta = {
  "message": "Unexpected socket close",
  "stack": "Error: Unexpected socket close\n    at Timeout.setTimeout (/usr/src/app/node_modules/nodemailer/lib/smtp-transport/index.js:189:31)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at unrefdHandle (timers.js:520:7)\n    at Timer.processTimers (timers.js:222:12)"
}

..................
```


```
user@hostname:~$ sudo docker logs k8s_alertmanager_alertmanager-584c4fcb59-7qft6_default_d47c392f-5a65-4905-b4b7-b0edf08ebb35_0
level=info ts=2021-11-18T13:33:36.498027171Z caller=main.go:174 msg="Starting Alertmanager" version="(version=0.15.1, branch=HEAD, revision=8397de1830f154535a31150f9262da0072d8725d)"
level=info ts=2021-11-18T13:33:36.498138944Z caller=main.go:175 build_context="(go=go1.10.3, user=root@efde7f9485ae, date=20180712-18:25:27)"
level=info ts=2021-11-18T13:33:36.502742394Z caller=main.go:320 msg="Loading configuration file" file=/etc/alertmanager/config.yml
level=info ts=2021-11-18T13:33:36.502795015Z caller=cluster.go:570 component=cluster msg="Waiting for gossip to settle..." interval=10m0s
level=info ts=2021-11-18T13:33:36.507253095Z caller=main.go:396 msg=Listening address=:9093
level=info ts=2021-11-18T13:34:36.503018922Z caller=cluster.go:579 component=cluster msg="gossip not settled but continuing anyway" polls=0 elapsed=59.999830923s
level=error ts=2021-11-18T13:40:54.148278125Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:40:54.148414991Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:45:54.148770584Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:45:54.149099339Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:50:54.149736354Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:50:54.149844742Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T13:55:54.150229731Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="context deadline exceeded"
level=error ts=2021-11-18T13:55:54.15066462Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="context deadline exceeded"
level=error ts=2021-11-18T14:00:54.153040339Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:00:54.15318812Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:05:54.154146299Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="context deadline exceeded"
level=error ts=2021-11-18T14:05:54.15427601Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="context deadline exceeded"
level=error ts=2021-11-18T14:10:54.1547301Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:10:54.154926268Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:15:54.155759899Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:15:54.156007768Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:20:54.161624022Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="context deadline exceeded"
level=error ts=2021-11-18T14:20:54.161705591Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="context deadline exceeded"
level=error ts=2021-11-18T14:25:54.163332609Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:25:54.163504314Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:30:54.163803922Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:30:54.163926295Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:35:54.16701424Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:35:54.167138359Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:40:54.168756763Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:40:54.169085803Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="Post http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage: dial tcp [::1]:9095: connect: cannot assign requested address"
level=error ts=2021-11-18T14:45:54.169611349Z caller=notify.go:332 component=dispatcher msg="Error on notify" err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
level=error ts=2021-11-18T14:45:54.169830051Z caller=dispatch.go:280 component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="unexpected status code 500 from http://localhost:9095/alert-handler/send-email-to-admin/?template=cluster-usage"
```

I found a timeout error in the log, however, I have used this configuration of smtp.163.com in other systems, and it has been verified to be usable.

--- a/src/alert-manager/src/cluster-utilization/send_alert.py
+++ b/src/alert-manager/src/cluster-utilization/send_alert.py
@@ -206,6 +206,9 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
             "cluster_usage": cluster_usage,
             "trigger_time": trigger_time,
         },
+        "annotations": {
+            "summary": "The weekly cluster usage has been reported, please check your email for details."
+        },
         "generatorURL": "alert/script"
     }
     alerts.append(alert)
@@ -226,6 +229,9 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
                 "status": job["status"],
                 "trigger_time": trigger_time,
             },
+            "annotations": {
+                "summary": "The weekly cluster usage has been reported, please check your email for details."
+            },
             "generatorURL": "alert/script"
         }
         alerts.append(alert)
@@ -241,6 +247,9 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
                 "resources_occupied": user["resources_occupied"],
                 "usage": user["usage"],
                 "trigger_time": trigger_time,
+            },
+            "annotations": {
+                "summary": "The weekly cluster usage has been reported, please check your email for details."
             },
             "generatorURL": "alert/script"
         }

--- a/src/alert-manager/src/cluster-utilization/send_alert.py
+++ b/src/alert-manager/src/cluster-utilization/send_alert.py
@@ -207,7 +207,7 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
             "trigger_time": trigger_time,
         },
         "annotations": {
-            "summary": "The weekly cluster usage has been reported, please check your email for details."
+            "summary": "The cluster usage has been reported, please check your email-box for details."
         },
         "generatorURL": "alert/script"
     }
@@ -230,7 +230,7 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
                 "trigger_time": trigger_time,
             },
             "annotations": {
-                "summary": "The weekly cluster usage has been reported, please check your email for details."
+                "summary": "The cluster usage has been reported, please check your email-box for details."
             },
             "generatorURL": "alert/script"
         }
@@ -249,7 +249,7 @@ def send_alert(pai_url: str, cluster_usage, job_usage, user_usage):
                 "trigger_time": trigger_time,
             },
             "annotations": {
-                "summary": "The weekly cluster usage has been reported, please check your email for details."
+                "summary": "The cluster usage has been reported, please check your email-box for details."
             },
             "generatorURL": "alert/script"
         }

